subsys/bluetooth/shell/bt.c

```diff
@@ -193,27 +193,38 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
+static void print_data_hex(const uint8_t *data, uint8_t len, bool reverse)
+{
+	shell_fprintf(ctx_shell, SHELL_NORMAL, "0x");
+	for (uint8_t i = 0; i < len; i++) {
+		shell_fprintf(ctx_shell, SHELL_NORMAL, "%02x", data[reverse ? len-1-i : i]);
+	}
+}
+
+static void print_data_set(uint8_t type, uint8_t initial_value_len, uint8_t set_value_len,
 			   const uint8_t *scan_data, uint8_t scan_data_len)
 {
-	uint8_t min_value_len = MIN(set_value_len, scan_data_len);
+	uint8_t idx = 0;
 
-	shell_fprintf(ctx_shell, SHELL_INFO, "%*sType 0x%02x: ",
-		      strlen(scan_response_label), "",
-		      type);
-	for (uint8_t i = 0U; i < (scan_data_len / min_value_len); i++) {
-		if (i > 0L) {
-			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
-		}
+	shell_fprintf(ctx_shell, SHELL_NORMAL, "%*sType 0x%02x: ",
+			  strlen(scan_response_label), "",
+			  type);
 
-		shell_fprintf(ctx_shell, SHELL_INFO, "0x");
-		for (uint8_t j = 0U; j < min_value_len; j++) {
-			shell_fprintf(ctx_shell, SHELL_INFO, "%02x",
-				      scan_data[i * min_value_len + j]);
-		}
+	if (initial_value_len > 0) {
+		print_data_hex(scan_data, initial_value_len, true);
+		shell_fprintf(ctx_shell, SHELL_NORMAL, ": ");
```

Comment:
The colon delimiter is already used for the Type. We should use a different delimiter.

Reply:
The UUID is a special part of the data. Maybe we can just do a space and then just have commas between data bytes/blocks for easier reading/parsing? E.g. 0x1234 0x56,0x78 ?

Reply:
I decided to go with your pattern and only comma separate the values (including the initial UUID) to make parsing easier (also removed the spaces between values)